lib/gears/timer.lua

```diff
@@ -215,14 +219,18 @@ function timer.start_new(timeout, callback)
     return t
 end
 
---- Create a timeout for calling some callback function.
--- This function is almost identical to `gears.timer.start_new`. The only difference
--- is that this does not prevent the callback function from being garbage
--- collected. After the callback function was collected, the timer returned
--- will automatically be stopped.
+--- Create a simple timer for calling the `callback` function continuously.
+--
+-- This function is almost identical to `gears.timer.start_new`. The only
+-- difference is that this does not prevent the callback function from being
+-- garbage collected.
```

Comment:
This new wording sounds as if something within the timer might cause the function to be garbage collected and if the user wants to keep references to the function elsewhere, they are supposed to account for that.
Whereas the functionality is the opposite of that, where the timer uses weak references so that the user may decide if/when the function is garbage collected.

Reply:
Hmmm... Yes, my version is probably oversimplified and too open for interpretation...